### PR TITLE
webgpu: add conv2d naive version for debugging

### PR DESCRIPTION
**Pull request by xhcao**
To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

- - -
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/6837)

### CODE REVIEW

**Comment by xhcao** [comment]
@qjia7 @axinging Please take a look, thank you

File: tfjs-backend-webgpu/src/kernels/Conv2D_impl.ts [hunk 1]
```diff
@@ -202,25 +208,39 @@ export function conv2DImpl({
...
-    {type: 'int32', data: [convInfo.dilationHeight, convInfo.dilationWidth]},
-    {type: 'int32', data: [dimAOuter]}, {type: 'int32', data: [dimBOuter]},
-    {type: 'int32', data: [dimInner]}
-  ];
-
-  const program = new Conv2DMMProgram(
-      convInfo, dimAOuter, dimBOuter, dimInner, hasBias, activation,
-      hasPreluActivationWeights);
+  if (useNaiveConv2d) {
+    dimensions = [
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/kernels/Conv2D_impl.ts:215]: Can we move the common `dimensions` part out of `if`? For `Conv2DNaiveProgram`, we only need to push `dimAOuter`, `dimBOuter`, and `dimInner`.

File: tfjs-backend-webgpu/src/conv2d_naive_webgpu.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+import {backend_util} from '@tensorflow/tfjs-core';
+
+import {activationFnSnippet, biasActivationSnippet} from './activation_util';
+import {getMainHeaderString as main, WebGPUProgram} from './webgpu_program';
+import {computeDispatch} from './webgpu_util';
+
+export class Conv2DNaiveProgram implements WebGPUProgram {
+  outputShape: number[];
+  shaderKey: string;
+  userCode: string;
```
[the remark this answers is not in the document]
Comment (xhcao) [reply tfjs-backend-webgpu/src/conv2d_naive_webgpu.ts]: Done

File: tfjs-backend-webgpu/src/kernels/Conv2D_impl.ts (hunk 1 above)
Comment (xhcao) [reply tfjs-backend-webgpu/src/kernels/Conv2D_impl.ts]: Done

File: tfjs-backend-webgpu/src/conv2d_naive_webgpu.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  dispatchLayout: {x: number[], y: number[], z: number[]};
+  dispatch: [number, number, number];
+  variableNames = ['x', 'W'];
+  uniforms =
+      'filterDims: vec2<i32>, pad: vec2<i32>, stride: vec2<i32>, dilation: vec2<i32>,';
+  workGroupSize: [number, number, number] = [4, 4, 8];
```
[the remark this answers is not in the document]
Comment (xhcao) [reply tfjs-backend-webgpu/src/conv2d_naive_webgpu.ts:32]: Jiajia, This naive kernel is only used as a reference of correctness, it does not take care of the performance, let us keep it simply and to be more cleaner. Is it OK?

[the remark this answers is not in the document]
Comment (qjia7) [reply tfjs-backend-webgpu/src/conv2d_naive_webgpu.ts:32]: SG

**Review by qjia7 — APPROVED** [review]
LGTM, thanks.

**Review by gyagp — APPROVED** [review]
LGTM

**Pull request merged**